packages/oui-pagination/src/index.spec.js

```diff
@@ -0,0 +1,336 @@
+            it("should have previous button disabled when not available", () => {
+                const element = TestUtils.compileTemplate(`
+                    <oui-pagination
+                        current-offset="1"
+                        page-size="25"
+                        total-items="100">
+                    </oui-pagination>
+                `);
+
+                expect(getPreviousButton(element).getAttribute("disabled")).toEqual("disabled");
+                expect(getNextButton(element).hasAttribute("disabled")).toBeFalsy();
+            });
+
+            it("should have next previous button disabled when not available", () => {
+                const element = TestUtils.compileTemplate(`
+                    <oui-pagination
+                        current-offset="76"
+                        page-size="25"
+                        total-items="100">
+                    </oui-pagination>
+                `);
+
+                expect(getPreviousButton(element).hasAttribute("disabled")).toBeFalsy();
+                expect(getNextButton(element).getAttribute("disabled")).toEqual("disabled");
+            });
+
+            it("should trigger on-change handler when next is clicked", () => {
+                const clickSpy = jasmine.createSpy("clickSpy");
+                const element = TestUtils.compileTemplate(`
+                    <oui-pagination
+                        current-offset="1"
+                        page-size="25"
+                        total-items="100"
+                        on-change="$ctrl.clickHandler($event)">
+                    </oui-pagination>
+                    `, {
+                        clickHandler: clickSpy
+                    });
+
+                angular.element(getNextButton(element)).triggerHandler("click");
+                expect(clickSpy).toHaveBeenCalledWith({
+                    offset: 26,
+                    pageSize: 25
+                });
+                expect(clickSpy.calls.count()).toEqual(1);
+            });
+
+            it("should trigger on-change handler when previous is clicked", () => {
+                const clickSpy = jasmine.createSpy("clickSpy");
+                const element = TestUtils.compileTemplate(`
+                    <oui-pagination
+                        current-offset="76"
+                        page-size="25"
+                        total-items="100"
+                        on-change="$ctrl.clickHandler($event)">
+                    </oui-pagination>
+                    `, {
+                        clickHandler: clickSpy
+                    });
+
+                angular.element(getPreviousButton(element)).triggerHandler("click");
+                expect(clickSpy).toHaveBeenCalledWith({
+                    offset: 51,
+                    pageSize: 25
+                });
+                expect(clickSpy.calls.count()).toEqual(1);
+            });
+        });
+
+        describe("Pages buttons", () => {
```

Comment:
It's not easy to read and it doesn't document the page buttons behavior.
e.g.
"should trigger on-change handler when a new page is selected" It should be separated in 3 distinct blocks
- Arrange template and mocks
- Act : Click a page button
- Assert :  Check on-change params

"should disable button of the current page" It should be separated in 3 distinct blocks
- Arrange template and mocks
- Act : Click a page button
- Assert :  Check attribute

...

When I read buttons[1], I don't know what it is. page1Button, page2Button could be easier to understand

Reply:
Sometime I like to organise a unit test to setup a small scenario. Maybe it's not the purpose of unit tests but we don't have integration tests, now... It could be rewritten as pure unit tests.

buttons[1] is no readable, please, you know that it's the second element of the buttons array described a few lines above ;)